lib/pages/tabPages/vendorsTabPage.dart

```diff
@@ -36,40 +39,64 @@ class _VendorsPageState extends State<VendorsPageWidget> {
   }
 
   void initPlatform() async {
+    //start loading timer :: 10s, if not done loading by then, display toast
+    _startLoadingTimer();
     //Intializing geoFire
     geo.initialize("Vendors_Location");
     try {
       var serviceStatus = await _locationService.checkGeolocationPermissionStatus();
-      print("Service status: $serviceStatus");
       if (serviceStatus == GeolocationStatus.granted) {
-          currentLocation = await _locationService.getLastKnownPosition(desiredAccuracy: LocationAccuracy.medium);
-          geo.queryAtLocation(currentLocation.latitude, currentLocation.longitude, 80.0);
-          geo.onKeyEntered.listen((data){
-            keyEntered(data);
-          });
-          geo.onKeyExited.listen((data){
-            keyExited(data);
-          });
-          _locationService.getPositionStream(LocationOptions(accuracy: LocationAccuracy.medium, distanceFilter: 400)).listen((Position result) async {
-            if (this.mounted){
-              setState(() {
-                currentLocation = result;
-              });
-              geo.updateLocation(currentLocation.latitude, currentLocation.longitude, 80.0);
-            }
+        currentLocation = await _locationService.getLastKnownPosition(desiredAccuracy: LocationAccuracy.medium); //this may be null! Thats ok!
+        geo.queryAtLocation(currentLocation.latitude, currentLocation.longitude, 80.0);
```

Comment:
If this is null, wouldn't you get a null pointer when you call currentLocation.latitude?